src/util/container_utils.h

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************\
+
+Module: Container utilities
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_CONTAINER_UTILS_H
+#define CPROVER_UTIL_CONTAINER_UTILS_H
+
+#include <set>
+
+/// Compute union of two sets
+///
+/// This function has complexity O(max(n1, n2)), with n1, n2 being the sizes of
+/// the sets of which the union is formed. This is in contrast to
+/// `target.insert(source.begin(), source.end())` which has complexity
+/// O(n2 * log(n1 + n2)).
+///
+/// \tparam T value type of the sets
+/// \tparam Compare comparison predicate of the sets
+/// \tparam Alloc allocator of the sets
+/// \param target first input set, will contain the result of the union
+/// \param source second input set
+/// \return true iff `target` was changed
+template <class T, class Compare, class Alloc>
+bool util_set_union(
```

Comment:
Suggest `_union_inplace`, since `set_union` would usually be a binary op that returns a new set.

Reply:
I just tried this, with `std::set<std::size_t>`'s, both sets being of equal size, and the sets containing random values. The runtimes (in microseconds) for the unions are as follows (union1 is the version in the PR). The leftmost column gives the sizes of the two sets being merged.

```
       union1     union2
100        39         75
  1K      368        698
 10K    3,358      6,704
100K   54,455    104,597
  1M  592,993  1,208,903
```

Reply:
Thank you!! I suppose that memory allocation is the dominating factor, which of course has to be done from scratch when using `std::set_union`. So I'd still suggest to rename the function to some variation involving "inplace", and then this PR should just be merged?!